Feature/modeling and vis and tracking (#1)
snowplow/consent-accelerator: content/_index.md

+++
title = "Snowplow Consent Tracking Accelerator"
menuTitle="Introduction"
chapter = false
weight = 1
+++

### Welcome to the Consent Tracking accelerator

> This accelerator will help you collect data from user interactions with your Consent Management Platform _(CMP)_ and their preferences, using the enhanced consent plugin from Snowplow.

Reviewer: ```suggestion
This accelerator will help you collect user interactions and preferences from your Consent Management Platform _(CMP)_ using the enhanced consent plugin from Snowplow.
```

> Data collection and user consent has become a main concern when talking about product, personalisation, marketing and ML-driven actions. At the same time, CMPs can provide a limited and siloed amount of data about user preferences and individual selection.

Reviewer: ```suggestion
User consent for data collection is vital when talking about product, personalisation, marketing and ML-driven actions. CMPs provide a limited amount of data about user preferences but with the enhanced consent plugin this can become a rich source of information to understand consent in your business.
```

By using Snowplow to collect consent preference, you can enrich your data model with user level, granular and context-rich selection data. Having user consent available in your central data store allows enriching your segments or ML models with this information and take action respectfully of a user's choices.

Here you will learn to:

- Set-up Snowplow Consent Tracking on your CMP (_example using Cookiebot_)
- Model your data using the Snowplow-Consent model
- Create a consent health check dashboard from your data in Streamlit

#### Who is this guide for?

- Data practitioners who would like to get familiar with the Snowplow enhanced-consent-plugin.
- Data practitioners who want to learn how to use the snowplow-consent dbt package and set-up tracking using their companies website or single page application, to gain insight from their customers’ behavioural data as quickly as possible.

#### Prerequisites

**Modeling and Visualisation**
- dbt CLI installed or dbt Cloud account available
  - New dbt project created and configured
- Python 3 Installed
- Snowflake or BigQuery account

> **Tracking and Enrichment**
> - Snowplow pipeline
> - Web app to implement tracking

Reviewer: ```suggestion
**Tracking and Enrichment**
- Snowplow pipeline
- Web app to implement tracking
- CMP Set up
```

Here you will learn how to:

- Set-up Snowplow Consent Tracking on your CMP (_example using Cookiebot_)
- Model your data using the Snowplow-Consent model
- Create a consent health check dashboard from your data in Streamlit

#### Who is this guide for?

- Data practitioners who would like to get familiar with the Snowplow enhanced-consent-plugin.
- Data practitioners who want to learn how to use the snowplow-consent dbt package and set-up tracking using their companies website or single page application, to gain insight from their customers’ behavioural data as quickly as possible.

***

#### What you will learn

In approximately 5 working hours you can achieve the following:

- **Track -** Set-up and deploy tracking to your website or single page application to allow consent tracking
- **Model -** Enable and run the consent module within the snowplow-web data model
- **Visualise -** Visualise the modeled data with Streamlit

{{<mermaid>}}
gantt
        dateFormat  HH-mm
        axisFormat %M
        section 1. Track

>         3h          :track, 00-00, 3h

Reviewer 2: The gantt chart isn't showing hours worked, it is only showing either 00 or 30 along the x axis. The other accelerators increase by the hours required e.g. 00    01      02      03     etc
Can we see if this is possible to do?

Reviewer: ```suggestion
        3h          :track, 00-00, 3m
```

Something like this should work. It doesn't like counting hours but minutes are fine. 

We're all good until we have an accelerator that takes more than 60 hours....

Reviewer 2: Let's hope we don't have a 60+ hour one. 
Thanks for sorting Jack

Reviewer: <img width="998" alt="Screenshot 2022-12-07 at 10 18 22" src="https://user-images.githubusercontent.com/87364579/206152510-a92c2288-f1c8-4735-88d7-45fa10c733d7.png">

Like this?

Author: yeah, but at least it is in hourly increments, I'm fine with either tbh
